src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -392,16 +384,39 @@ protected void parseCreateField(ParseContext context) throws IOException {
 
     protected void parseCreateField(ParseContext context, int dimension) throws IOException {
 
-        if (!KNNSettings.isKNNPluginEnabled()) {
-            throw new IllegalStateException("KNN plugin is disabled. To enable " + "update knn.plugin.enabled setting to true");
+        validateIfKNNPluginEnabled();
+        validateIfCircuitBreakerTriggered();
```

Comment:
This part will be common with Lucene field mapper, extract it to a method to avoid code duplication